Java/ImplementingBST.java

```diff
@@ -0,0 +1,167 @@
+		System.out.println("Enter element to search");
+		int ele = sc.nextInt();
+		boolean found = searchElementInBST(root, ele);
+		if (found) {
+			System.out.println("Found " + ele);
+		} else {
+			System.out.println("Did not Find " + ele);
+		}
+		System.out.println("---Delete Implementation---");
+		System.out.println("Enter element to deleted");
+		ele = sc.nextInt();
+		root = deleteElementFromBST(root, ele);
+		System.out.println("---After deletion, BST looks like---");
+		inOrderBST(root);
+
+	}
+	//delete node from BST
+	private static TreeNode deleteElementFromBST(TreeNode root, int ele) {
+		// TODO Auto-generated method stub
+		if (root == null) {
+			return root;
+		}
+		if (ele < root.value) {
+			root.left = deleteElementFromBST(root.left, ele);
+		} else if (ele > root.value) {
+			root.right = deleteElementFromBST(root.right, ele);
+		} else {
+			if (root.left == null)
+				return root.right;
+			else if (root.right == null)
+				return root.left;
+			root.value = minValue(root.right);
+			root.right = deleteElementFromBST(root.right, root.value);
+		}
+
+		return root;
+	}
+	
+	//finding minimum successor
+	private static int minValue(TreeNode root) {
+		// TODO Auto-generated method stub
+		  int minv = root.value; 
+	        while (root.left != null) 
+	        { 
+	            minv = root.left.value; 
+	            root = root.left; 
+	        } 
+	        return minv; 
+	}
+	
+	//search element in BST
+	private static boolean searchElementInBST(TreeNode root, int ele) {
+		// TODO Auto-generated method stub
+		if (root == null) {
+			return false;
+		}
+		if (ele == root.value) {
+			return true;
+		}
+
+		if (ele < root.value) {
+			return searchElementInBST(root.left, ele);
+		}
+
+		return searchElementInBST(root.right, ele);
+	}
+	//InOrder Traversal to print nodes
+	private static void inOrderBST(TreeNode root) {
+		// TODO Auto-generated method stub
+		if (root == null) {
+			return;
+		}
+		inOrderBST(root.left);
+		System.out.print(root.value + " ");
+		inOrderBST(root.right);
+
+	}
+	//Insert to BST
+	private static TreeNode insertToBST(TreeNode root, int ele) {
+		// TODO Auto-generated method stub
+		if (root == null) {
+			root = new TreeNode(ele);
+			return root;
+		}
+		if (ele < root.value) {
+			root.left = insertToBST(root.left, ele);
+		} else {
+			root.right = insertToBST(root.right, ele);
+		}
+		return root;
+	}
+
+}
+
+/*
+ * ---Insert Implementation---
+How many elements do you want to insert
+5
+Enter elements to BST
+10
+Successfully inserted 10
+BST looks like 
+10 
+9
+Successfully inserted 9
+BST looks like 
+9 10 
+5
+Successfully inserted 5
+BST looks like 
+5 9 10 
+20
+Successfully inserted 20
+BST looks like 
+5 9 10 20 
+12
+Successfully inserted 12
+BST looks like 
+5 9 10 12 20 
+
+---Searching Impelementation--
+Enter element to search
+12
+Found 12
+---Delete Implementation---
+Enter element to deleted
+9
+---After deletion, BST looks like---
+5 10 12 20 
+*/
```

Comment:
Please add time (for each operation) and space complexity as well.
Otherwise, nice work!
Ready to merge as well !
Let me know once you are done @mephi007

Reply:
hey @sukritishah15, I have added Time complexity and Space Complexity into the code... Can you please review it once?